src/main/groovy/io/github/cosmicsilence/scalafix/ScalafixTask.groovy

```diff
@@ -17,7 +18,8 @@ class ScalafixTask extends SourceTask {
 
     @InputFile
     @Optional
-    final RegularFileProperty configFile = project.objects.fileProperty()
+    final RegularFileProperty configFile = GradleCompat.isVersion4(project)?
+            project.layout.fileProperty(): project.objects.fileProperty()
```

Comment:
the idea here is that you could use `configFile = GradleCompat.fileProperty(project)`